include/_internal_s2n_bignum_arm.h

```diff
@@ -0,0 +1,45 @@
+#ifdef __APPLE__
+#   define S2N_BN_SYMBOL(NAME) _##NAME
+#else
+#   define S2N_BN_SYMBOL(name) name
+#endif
+
+#define S2N_BN_SYM_VISIBILITY_DIRECTIVE(name) .globl S2N_BN_SYMBOL(name)
+
+#ifdef S2N_BN_HIDE_SYMBOLS
+#   ifdef __APPLE__
+#      define S2N_BN_SYM_PRIVACY_DIRECTIVE(name) .private_extern S2N_BN_SYMBOL(name)
+#   else
+#      define S2N_BN_SYM_PRIVACY_DIRECTIVE(name) .hidden S2N_BN_SYMBOL(name)
+#   endif
+#else
+#   define S2N_BN_SYM_PRIVACY_DIRECTIVE(name)  /* NO-OP: S2N_BN_SYM_PRIVACY_DIRECTIVE */
+#endif
+
+#ifdef __APPLE__
+#   define S2N_BN_FUNCTION_TYPE_DIRECTIVE(name) /* Not used in Mach-O */
+#else
+#   define S2N_BN_FUNCTION_TYPE_DIRECTIVE(name) .type name, %function
+#endif
+
+// Variants of instructions including CFI (call frame information) annotations
+
+#define CFI_START .cfi_startproc
+#define CFI_RET ret ; .cfi_endproc
+
+#define CFI_BL(target) bl target
```

Comment:
Do we need these function call wrapper macros? it seems like they don't do anything interesting?

Reply:
No, that's true - I must have thought there would be some annotation to add later. I am still wondering if I might need something to pacify `objtool`, as suggested in its documentation, but perhaps the real issue is that the subroutine itself isn't properly annotated or is not considered as a distinct function from a CFI point of view (as was, provisionally at least, intended). I'll have to look more at the pointers you sent above.

> 11. file.o: warning: unannotated intra-function call
> 
>     This warning means that a direct call is done to a destination which
>     is not at the beginning of a function. If this is a legit call, you
>     can remove this warning by putting the ANNOTATE_INTRA_FUNCTION_CALL
>     directive right before the call.
>